modules/ROOT/pages/syntax/expressions.adoc

```diff
@@ -1232,6 +1232,187 @@ RETURN COLLECT {
 1+d|Rows: 1
 |===
 
+[[type-predicate-expressions]]
+== Type predicate expressions
+
+_This feature was introduced in Neo4j 5.9._
+
+A type predicate expression can be used to verify the type of a variable, literal, property or other Cypher expression.
+
+[source, syntax, role="noheader", indent=0]
+----
+<expr> IS :: <TYPE>
+----
+
+Where `<expr>` is any Cypher expression and `<TYPE>` is one of the following types:
+
+* `BOOLEAN`
+* `STRING`
+* `INTEGER`
+* `FLOAT`
+* `DATE`
+* `LOCAL TIME`
+* `ZONED TIME`
+* `LOCAL DATETIME`
+* `ZONED DATETIME`
+* `DURATION`
+* `POINT`
+
+Allowed syntax variations of these types are listed xref::values-and-types/property-structural-constructed.adoc#types-synonyms[here].
+
+.Query
+[source, cypher]
+----
+UNWIND [42, true, 'abc'] AS val
+RETURN val, val IS :: INTEGER AS isInteger
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| +val+ | +isInteger+
+| +42+ | +true+
+| +true+ | +false+
+| +'abc'+ | +false+
+2+d|Rows: 3
+|===
+
+[[type-predicate-expressions-not]]
+=== Type predicate expressions with NOT
+
+It is also possible to verify that a Cypher expression does not have a certain type, using `IS NOT ::`.
+
+.Query
+[source, cypher]
+----
+UNWIND [42, true, 'abc'] AS val
+RETURN val, val IS NOT :: STRING AS notString
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="2*<m"]
+|===
+| +val+ | +notString+
+| +42+ | +true+
+| +true+ | +true+
+| +'abc'+ | +false+
+2+d|Rows: 3
+|===
+
+[[type-predicate-expressions-null]]
+=== Type predicate expressions for null
+All Cypher types are considered to include the `null` value.
+Therefore, `IS ::` will return `true` for all expressions evaluating to `null`, regardless of the type.
+
+.Query
+[source, cypher]
+----
+RETURN toFloat(null) IS :: BOOLEAN AS isBoolean
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="1*<m"]
+|===
+| +isBoolean+
+| +true+
+1+d|Rows: 1
+|===
+
+Likewise, `IS NOT ::` will return `false` for all expressions evaluating to `null`, regardless of the type.
+
+.Query
+[source, cypher]
+----
+RETURN null + 1 IS NOT :: DATE AS isNotDate
+----
+
+.Result
+[role="queryresult",options="header,footer",cols="1*<m"]
+|===
+| +isNotDate+
+| +false+
+1+d|Rows: 1
+|===
+
+[[type-predicate-expressions-properties]]
+=== Type predicate expression for properties
+Type predicate expressions can also be used to filter out nodes or relationships with properties of a certain type.
+
+Assume that we have a graph containing the following nodes:
```

Comment:
Do we normally use personal pronouns in the docs?

Reply:
Good catch, I will check how it is written above